File: day12_today_i_will_finish_strings/zeromatrix.go
package main

import "fmt"
// the question asks if the  element in an m*n matrix is 0, if yes then changes the entire 
// row and columns of the element to 0.


func checkForZero(matrix [][]int)  [][]int{
	rows:=make([]bool,len(matrix))
	cols:=make([]bool,len(matrix))

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < len(matrix); j++ {
			if matrix[i][j]==0 {
				rows[i]=true
				cols[j]=true		
			}
		}
	}
	for row := 0; row < len(rows); row++ {
		if rows[row] {
			zeroRows(matrix,row)
		}
	}

	for col := 0; col < len(rows); col++ {
		if cols[col] {
			zeroCols(matrix,col)
		}
	}
	return matrix

}

func zeroRows(matrix [][]int, row int)  {
	for i := 0; i < len(matrix[0]); i++ {
		matrix[row][i]=0
	}
}

func zeroCols(matrix [][]int, cols int)  {
	for i := 0; i < len(matrix); i++ {
		matrix[i][cols]=0
	}
}


func main(){
	arr:=[][]int{{1,0,3},
	{4,5,6},
	{7,8,9}}
fmt.Println(checkForZero(arr))
}

File: day12_today_i_will_finish_strings/zeromatrix.cpp
#include<iostream>
#include<vector>
using namespace std;

void zeroRows(vector<vector<int>> matrix,int row) {

    for (int i = 0; i < matrix[0].size(); i++)
    {
        matrix[row][i]=0;
    }
    
}

void zeroCols(vector<vector<int>> matrix,int col) {

    for (int i = 0; i < matrix.size(); i++)
    {
        matrix[i][col]=0;
    }
    
}

vector<vector<int>> checkForZero(vector<vector<int>> matrix){

    vector<int> rows;
    vector<int> cols;

    for (int i = 0; i < matrix.size(); i++)
    {
        for (int j = 0; j < matrix.size(); j++)
        {
            if (matrix[i][j]==0)
            {
                rows[i]=true;
                cols[i]=true;
            }
            
        }
        
    }
    for (int row = 0; row < rows.size(); row++)
    {
        if (rows[row])
        {
           zeroRows(matrix,row);
        }
        
    }

    for (int col = 0; col < rows.size(); col++)
    {
        if (cols[col])
        {
           zeroCols(matrix,col);
        }
        
        /* code */
    }
    return matrix;
}


int main(int argc, char const *argv[])
{
    vector<vector<int> > vec{ { 1, 2, 3 }, 
                         { 4, 5, 6 }, 
                         { 7, 8, 9} }; 
    auto matrix=checkForZero(vec);
    for (int i = 0; i < matrix.size(); i++) { 
        for (int j = 0; j < matrix[i].size(); j++) 
            cout << matrix[i][j] << " "; 
        cout << endl; 
    }
    return 0;
}
